uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/main/java/org/uberfire/ext/wires/core/grids/client/widget/grid/renderers/columns/impl/ColumnRenderingStrategyMerged.java

```diff
@@ -54,187 +56,182 @@ public Group render(final GridColumn<?> column,
+            final MultiPath bodyGrid = theme.getBodyGridLine();
+            for (int rowIndex = minVisibleRowIndex; rowIndex <= maxVisibleRowIndex; rowIndex++) {
+                final double y = visibleRowOffsets.get(rowIndex - minVisibleRowIndex) - visibleRowOffsets.get(0);
+                final GridRow row = model.getRow(rowIndex);
+
+                if (!row.isMerged()) {
+                    //If row doesn't contain merged cells just draw a line across the visible body
+                    bodyGrid.M(x, y + 0.5)
+                            .L(x + columnWidth, y + 0.5);
+                } else if (!row.isCollapsed()) {
+                    //If row isn't collapsed just draw a line across the visible body at the top of the merged block
+                    final GridCell<?> cell = model.getCell(rowIndex,
+                                                           columnIndex);
+
+                    if (cell == null || cell.getMergedCellCount() > 0) {
+                        //Draw a line-segment for empty cells and cells that are to have content rendered
+                        bodyGrid.M(x, y + 0.5)
+                                .L(x + columnWidth, y + 0.5);
+                    } else if (isCollapsedRowMultiValue(model,
+                                                        column,
+                                                        cell,
+                                                        rowIndex)) {
+                        //Special case for when a cell follows collapsed row(s) with multiple values
+                        bodyGrid.M(x, y + 0.5)
+                                .L(x + columnWidth, y + 0.5);
+                    }
                 }
             }
-        }
 
-        //Grid lines - vertical
-        final int columnIndex = model.getColumns().indexOf(column);
-        if (columnIndex < model.getColumnCount() - 1) {
-            bodyGrid.M(columnWidth + 0.5,
-                       0).L(columnWidth + 0.5,
-                            columnHeight);
-        }
+            //- vertical
+            if (columnIndex < model.getColumnCount() - 1) {
+                bodyGrid.M(x + columnWidth + 0.5, 0)
+                        .L(x + columnWidth + 0.5, columnHeight);
+            }
 
-        //Column content
-        final Group columnGroup = new Group();
-        int iterations = 0;
-        for (int rowIndex = minVisibleRowIndex; rowIndex <= maxVisibleRowIndex; rowIndex++) {
+            parent.add(bodyGrid);
+        });
 
-            iterations++;
-            if (iterations > 1000) {
-                break;
-            }
+        //Column content
+        commands.add((GridRenderer.RenderBodyGridContentCommand) (parent) -> {
+            final Group columnGroup = new Group().setX(x);
+            int iterations = 0;
+            for (int rowIndex = minVisibleRowIndex; rowIndex <= maxVisibleRowIndex; rowIndex++) {
+
+                iterations++;
+                if (iterations > 1000) {
```

Comment:
Uf, where comes this constant 1000 from? Should we at least use `rowIndex - minVisibleRowIndex > 1000`